### PR TITLE
interface

### PR DESCRIPTION
**Pull request by MusinaPolina**

### CODE REVIEW

File: src/commonMain/kotlin/org/kosat/Solver.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package org.kosat
+
+typealias Lit = Int
+
+interface Solver {
```
Comment (Lipen) [inline src/commonMain/kotlin/org/kosat/Solver.kt:5]: This interface is meaningless, because you have exactly ONE implementation of the `Solver` — namely, `CDCL`. (No, DPLL should not be made another inheritor). Additionally, this interface contains methods for which you do not have an implementation at the moment (and probably never will with this exact interface), e.g., `getModel` or `interrupt`.
It is better to just implement you algorithms as a simple class (`CDCL`), without bothering yourself with an interface.
Another point for this: think about a hypothetical user (you can substitute "hypothetical user" with "yourself, but in the future/in another module (e.g. frontend)") of your package, including the class `CDCL` and the interface `Solver`. Do you really think the user is going choose to interact with CDCL algorithms and heuristics through the cutted-down and incomplete interface `Solver` which lacks the flexibility, or will it just choose to use `CDCL` class (with all the internal methods available) directly? I bet the latter.

TLDR: do not be interface-centric here.
